### ISSUE TITLE
New Backend Repo + Transition from Ruby to Python

### ISSUE THREAD

**Issue by nb333** [maridco/marid-backend#1]
![Hello!](https://cloud.githubusercontent.com/assets/5732914/6654079/9bf4ae4e-ca7b-11e4-9094-af7c1c7ac3f5.png)

Welcome to our new backend repo!

@joshghent @rich-iannone @naveen09 @johnmcconnell @tomohitoy @jpavlo @sandlbn @sant0sh @blakedietz @iwebhub @nitish1099 @aditya91 @blelump @danigonlinea @ZDroid @feenr @HHammond @jakeoverall @kendopunk @kingtak @nav-mike @thegeektets @titannguyen 

After some discussions with @joshghent, I've decided to removed Marid from Basecamp. From experience, projects have gone best when GitHub is used as the primary source of issue tracking, communicating, sharing ideas and assigning tasks. Plus, it just makes sense, since Marid is an open source project.

For those who haven't been involved in recent conversations, this new backend repo is for us to start the backend from scratch with Python (instead of Ruby). The current state of the Ruby backend can be found [here](https://github.com/maridco/marid), please disregard the frontend files in this repo as they're out-of-date. @sandlbn created the Ruby backend and wants to help with our transition to Python - if you have any questions about it, please ask him. The current frontend repo can be found [here](https://github.com/maridco/marid-frontend).

I'm also attaching the OpenFaux Server-Side Team (Python Devs) cc: @Sebastiani @schumannd @sant0sh @Wingie @boxtown @arunenigma @mouuff @thiagopa @metude @phillipbroberts @indradhanush @dongsam @quanticle @bangarang @caseybecking @stevenschmatz @jhg

Welcome to our project, Marid. We're working on an open source music platform that beautifully showcases the works of independent and unsigned artists; this includes their music, videos, lyrics, and images - like a portfolio. To make it even better, we're allowing music producers, engineers, artists, and songwriters to all join the platform. Marid provides them with the tools and insights they need to express themselves and potentially get signed, build a fan base, or just team up with other amazing talent... in other words, ultimately succeed; for every user that views their content, we have a built-in dashboard for them to view those stats: where, what, when, why, and who is viewing your work. Utilizing D3.js visualizations, users can select by country and even zoom in on state/province levels on the map and break down the areas in the world that love your work the most and so forth. Having those detailed of statistics allow us to feature on our homepage: the top artists near you, music that you should like, songs you've recently listened to, and what's generally popular right now - which also ties into our idea for the Python recommendation system @sant0sh has been working on. I would be highly grateful if any of you guys could help with our transition from Ruby to Python. If you have any questions about Marid, feel free to ask us.

I look forward to taking Marid to the next level with you guys!

Nathan

**Comment by joshghent** [comment]
Welcome everyone, nice to have you part of the project.

If anyone has any questions about the front-end repo, either raise an issue or email me (me@joshghent.com).

**Comment by HHammond** [comment]
Now that this project is moving to Python, what frameworks will it use? My vote is on Flask with Jinja2 and Peewee. I've found those tools to be powerful and highly productive while also being flexible. SQLAlchemy is popular too but I haven't tried it out so I can't comment on it's ease of use.

**Comment by phillipbroberts** [comment]
+1 Flask

SQLAlchema is a nice abstraction layer, but honestly unnecessary. If you
are used to Active Record etc, it might be a nice transition.

**Comment by sandlbn** [comment]
Why not django, simple and fast, most of the things already in the box. flask is good but after when you will need to add another 3rd party application is very messy.

**Comment by HHammond** [comment]
I don't find adding additional frameworks to flask to be very difficult. I prefer Peewee's ORM to Django's because I feel like you get much more control over queries and for our purposes (generating statistics for dashboards) is much easier to query. Aggregates and group by functions are much more straight forward in Peewee. 

I also really like Jinja2 so that email templates and application templates can be written using the same framework. 

I do think the major question should be which choice well get this project up and I'm quicker. If more devs know django and we can start faster then that is a huge advantage.

**Comment by joshghent** [comment]
@HHammond I agree with @sandlbn here. Using lots of 3rd party applications means there is more points of failure in our stack. A chain is only as strong as it's weakest link. Less software, less stuff to go wrong. 

I'm not saying Django is better, simple that it would not be a good idea to be using lots of different applications.

**Comment by HHammond** [comment]
What happened to all the documents that were on BaseCamp? Is there a repository where they were reuploaded?

**Comment by nb333** [comment]
@HHammond I've exported the documents to HTML from Basecamp. A lot of the info is out-of-date, so I'm going to create a wiki or gist and outline all of the important details from it, then we can collaborate on it.

**Comment by nb333** [comment]
@HHammond Here's the wiki: https://github.com/maridco/marid-backend/wiki

I've created a couple pages - the details came from our Basecamp:
- **Differentiators** - outlines how we're different from the competition
- **Pages** - outlines the purpose of each page
- **Tasks** - outlines each task (some of these will need to be redone, especially on the backend, as we're transitioning to Python)

I would be highly grateful if someone can help clean up the tasks and get things in order, especially on the backend.

**Comment by joshghent** [comment]
@nb333 I'll get on that in the morning.

**Comment by sebastiani** [comment]
I wouldn't use SQLAlchemy if we are expecting to scale up...

On Tue, Mar 17, 2015 at 10:28 PM, Josh Ghent notifications@github.com
wrote:

> @nb333 https://github.com/nb333 I'll get on that in the morning.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/maridco/marid-backend/issues/1#issuecomment-82690680.

## 

Sebastiani Aguirre Navarro
Bachelor Student in Electrical Engineering and Applied Mathematics at
University of Puerto Rico
Tel. 787-469-4138
